src/hw_mainline.js

```diff
@@ -62,17 +62,26 @@ var readPinMux = function (pin, mode, callback) {
         if (err) {
             mode.err = 'readPinctrl error: ' + err;
             if (debug) winston.debug(mode.err);
-            callback(mode);
+            if (callback.length == 1)
```

Comment:
When called via RPC, is the length transferred properly? I don't remember the exact process, but want to make sure this works in both the local and RPC cases.

Reply:
I did not test it previously via RPC , now when i tested : the nodestyle callback does not work properly via RPC , the previous style of callback work without any issues, will look more into this.

Reply:
Both the [RPC client](https://github.com/jadonk/bonescript/blob/451348bc/src/bonescript.js#L73-L77) and [RPC server](https://github.com/jadonk/bonescript/blob/451348bc/src/socket_handlers.js#L60-L91) make hardcoded use of the old callback style. The client will need to be fixed to accept both types of callback handler and do the necessary translation. If the old callback style is to be deprecated, the server will need to be fixed to use node-style callbacks.

BTW, while taking a quick peek at this RPC code, I spotted an obvious memory leak:
```
dev:~/beagle/bonescript$ grep -r oneshot src/
src/socket_handlers.js:                // TODO: consider setting 'oneshot'
src/bonescript.js:    if (data.oneshot) delete _bonescript._callbacks[data.seq];
```
In other words, callback handlers are never removed from the client's bookkeeping and the _bonescript._callbacks array will keep growing forever.